spackbot/handlers/reviewers.py

```diff
@@ -263,11 +264,14 @@ async def add_reviewers(event, gh):
             else:
                 logger.info(f"Adding collaborators: {non_reviewers}")
                 for user in non_reviewers:
-                    await gh.put(
-                        members_url,
-                        {"member": user},
-                        data={"role": "member"},
-                    )
+                    try:
+                        await gh.put(
+                            members_url, {"member": user}, data={"role": "member"}
+                        )
+                    except BadRequest as e:
+                        logger.warning(
+                            f"Skipping adding member {user}, likely already added: {e}"
+                        )
```

Comment:
This should check `e.status_code` and log as above if it's 404, but raise if we get other 4xx errors.  e.g., `422` is documented as "attempt to add org to team", and if we get `422`, it's likely a logic error we need to fix, not something we want to ignore.  Same for `403` -- `403` is a setup error and it would mean that Spackbot didn't have enough permissions on the repo.

Reply:
yep that works too - updated.